src/registry/generate.rs

```diff
@@ -402,4 +421,101 @@ mod tests {
 
         assert_eq!(rust_files, expected_rust_files);
     }
+
+    #[test]
+    fn test_registry_generate_with_symbolic_link_cases() {
```

Comment:
The main issue I’m facing is having a unit test in the main crate that depends on test data from another crate (`crates/weaver_codegen_test`). I think it would be better to have this symbolic link in the `data` directory of the main crate to avoid this kind of dependency, which might cause problems at some point.

Reply:
Thank you for pointing that out! The reason I created the symbolic link in `crates/weaver_codegen_test` is because, based on the function [test_registry_generate](https://github.com/Leo6Leo/weaver/blob/38154d64478182cc3287aeb5766deb1e8392868c/src/registry/generate.rs#L245), it appears to rely on the registry located in `crates/weaver_codegen_test/semconv_registry/`.

I think this indicates that the test depends on the `weaver_codegen_test` crate to access the registry. To ensure the required dependencies are properly linked, I created the symbolic link there to align with this structure.

But I have created the commit to reflect the suggestion you recommend, PTAL https://github.com/open-telemetry/weaver/pull/468/commits/f985907f7dd92264d1211d453fbc681832287b75 @lquerel